src/lib/storage/table.cpp

```diff
@@ -1,84 +1,106 @@
 #include "table.hpp"
 
+#include "resolve_type.hpp"
 #include "utils/assert.hpp"
+#include "value_segment.hpp"
 
 namespace opossum {
 
-Table::Table(const ChunkOffset target_chunk_size) {
-  // Implementation goes here
+Table::Table(const ChunkOffset target_chunk_size) : _target_chunk_size(target_chunk_size) {
+  create_new_chunk();
 }
 
 void Table::add_column_definition(const std::string& name, const std::string& type, const bool nullable) {
-  // Implementation goes here
+  _column_names.emplace_back(name);
+  _column_types.emplace_back(type);
+  _column_nullables.emplace_back(nullable);
 }
 
 void Table::add_column(const std::string& name, const std::string& type, const bool nullable) {
-  // Implementation goes here
-  Fail("Implementation is missing.");
+  DebugAssert(_chunks.size() <= 1, "It is only possible to add new columns to an empty table.");
+  DebugAssert(_column_names.size() < std::numeric_limits<ColumnCount>::max(), "Column limit is already reached.");
+  resolve_data_type(type, [this, nullable](const auto data_type_t) {
+    using ColumnDataType = typename decltype(data_type_t)::type;
+    const auto value_segment = std::make_shared<ValueSegment<ColumnDataType>>(nullable);
+    this->_chunks.at(0)->add_segment(value_segment);
+  });
+  add_column_definition(name, type, nullable);
 }
 
 void Table::create_new_chunk() {
-  // Implementation goes here
+  DebugAssert(_chunks.size() < std::numeric_limits<ChunkID>::max(), "Chunk limit is already reached.");
+
+  const auto chunk = std::make_shared<Chunk>();
+  for (auto i = 0; i < column_count(); i++) {
+    resolve_data_type(_column_types.at(i), [this, i, &chunk](const auto data_type_t) {
+      using ColumnDataType = typename decltype(data_type_t)::type;
+      const auto value_segment = std::make_shared<ValueSegment<ColumnDataType>>(_column_nullables.at(i));
+      chunk->add_segment(value_segment);
+    });
+  }
+  _chunks.emplace_back(chunk);
 }
 
 void Table::append(const std::vector<AllTypeVariant>& values) {
-  // Implementation goes here
-  Fail("Implementation is missing.");
+  if (_chunks.back()->size() >= target_chunk_size()) {
+    create_new_chunk();
+  }
```

Comment:
If the size is actually larger than the _target_chunk_size, the program is already in an inconsistent state. I'd argue you should check for equality 
```suggestion
  if (_chunks.back()->size() == target_chunk_size()) {
    create_new_chunk();
  }
```

Reply:
Probably a better approach is to add this condition as a `DebugAssert`, which might lead to much easier debugging in the future.

Reply:
I don't quite get what the assertion you suggest should do. After all, we need an if condition here to conditionally create a new chunk.